File: smartpark/models/Booking.js
const mongoose = require("mongoose");

const bookingSchema = new mongoose.Schema({
  user_id: {
    type: mongoose.Schema.Types.ObjectId,
    ref: "User",
    required: true
  },
  lot_id: {
    type: mongoose.Schema.Types.ObjectId,
    ref: "ParkingLot",
    required: true
  },
  payment_id: String,
  order_id: String,
  payment_status: {
    type: String,
    enum: ["pending", "paid", "failed"],
    default: "pending"
},
  slot_id: {
    type: mongoose.Schema.Types.ObjectId,
    ref: "Slot",
    required: true
  },
  start_time: {
    type: Date,
    default: Date.now
  },
  end_time: Date,
  total_price: Number,
  status: {
    type: String,
    enum: ["active", "completed", "cancelled"],
    default: "active"
  }
}, { timestamps: true });

module.exports = mongoose.model("Booking", bookingSchema);

File: smartparkapp/lib/models/booking.dart
class Booking {
  final String id;
  final String lotId;
  final String slotId;
  final String paymentStatus;
  final DateTime startTime;
  final DateTime? endTime;
  final double totalPrice;

  Booking({
    required this.id,
    required this.lotId,
    required this.slotId,
    required this.paymentStatus,
    required this.startTime,
    this.endTime,
    required this.totalPrice,
  });

  factory Booking.fromJson(Map<String, dynamic> json) {
    return Booking(
      id: json['_id'] as String,

      lotId: json['lot_id'] is Map
          ? json['lot_id']['_id'] as String
          : json['lot_id'] as String,

      slotId: json['slot_id'] is Map
          ? json['slot_id']['_id'] as String
          : json['slot_id'] as String,

      paymentStatus: json['payment_status'] as String? ?? 'pending',

      startTime: DateTime.parse(json['start_time'] as String),

      endTime: json['end_time'] != null
          ? DateTime.parse(json['end_time'] as String)
          : null,

      totalPrice: (json['total_price'] as num?)?.toDouble() ?? 0.0,
    );
  }
}
